src/libraries/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs

```diff
@@ -510,14 +510,10 @@ private static void FillNonZeroBytes(Span<byte> data)
                 RandomNumberGenerator.Fill(data);
 
                 // Find the first zero in the remaining portion.
-                int indexOfFirst0Byte = data.Length;
-                for (int i = 0; i < data.Length; i++)
+                int indexOfFirst0Byte = data.IndexOf((byte)0);
```

Comment:
Btw, technically, `IndexOf()` could use `IsKnownConstant` and redirect work to `IndexOfNullByte` specialized helper that we already have for better perf 🙂

Reply:
@jkotas a random question - [IndexOfNullByte](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs#L343-L346) API only accepts `(ref byte searchSpace)` argument and tells user not to worry about Length as it's going to align it and will never cross the page boundary - from my understanding it means it's designed to read more than needed. With `ref byte`.  Does it smell like a potential gc hole?

Reply:
The implementation assumes that `Vector128/256.LoadUnsafe` is going to be done atomically, without the JIT ever materializing byrefs that points into the middle of the `Vector`. Can this assumption be broken by JIT stress modes? 

There seems to be other subtle corner-case bugs. For example: The GC can move the object right after this alignment check https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs#L421 . It means that the subsequent `Vector256.LoadUnsafe` can be misaligned and it can crash if it falls on page boundary and the next page is unmapped.

Reply:
> redirect work to IndexOfNullByte specialized helper that we already have for better perf if needle is a constant zero

IndexOfNullByte does not take length. You would have to add an extra length argument.

Reply:
> Can this assumption be broken by JIT stress modes?

It would be legal, as vectors don't need to be loaded atomically, though I don't believe any current transformation would break up a vector if the source code didn't contain something promotable.

Reply:
Filed https://github.com/dotnet/runtime/issues/81332 to handle this